docs/root/version_history/current.rst

```diff
@@ -38,6 +38,14 @@ Minor Behavior Changes
 * http: the behavior of the ``present_match`` in route header matcher changed. The value of ``present_match`` is ignored in the past. The new behavior is ``present_match`` performed when value is true. absent match performed when the value is false. Please reference :ref:`present_match
   <envoy_v3_api_field_config.route.v3.HeaderMatcher.present_match>`.
 * listener: added an option when balancing across active listeners and wildcard matching is used to return the listener that matches the IP family type associated with the listener's socket address. Any unexpected behavioral changes can be reverted by setting runtime guard ``envoy.reloadable_features.listener_wildcard_match_ip_family`` to false.
+* listener: added the :ref:`enable_reuse_port <envoy_v3_api_field_config.listener.v3.Listener.enable_reuse_port>`
+  field and changed the default for reuse port from false to true, as the feature is now well
+  supported on the majority of production kernels in use. The default change is aware of hot
+  restart, as otherwise the change would not be backwards compatible between restarts. This means
+  that hot restarting on to a new binary will retain the default of false until the binary undergoes
+  a full restart. To retain the previous behavior, either explicitly set the new configuration
+  field to false, or set the runtime feature flag `envoy.reloadable_features.listener_reuse_port_default_enabled`
+  to false.
```

Comment:
I agree with the basic sequencing and WKT/runtime philosophy here. One thing I wonder though is whether technically we have to go through a deprecation cycle (i.e. minor version) before flipping the default. The reason is that we probably want to preserve the following property in a stable API:

"Any management server sending configuration X to some client at version N should continue to have the same behavior when sending X to some client at N+1 (modulo our versioning policy)"

Basically,  management servers should generally set explicitly WKT values if they don't want client-dependent behavior, but we can't add surprise new WKT fields with defaults that override other fields unless we want to give up ^^.

This becomes important when thinking about xDS-as-a-service management servers. What is their migration path?

Interested to hear what other @envoyproxy/api-shepherds have to say.

Reply:
^ is basically why I wrote it this way, but I agree that we should explicitly codify this if we agree with this approach.

Reply:
Oh sorry I missed what you meant about keeping the default as is for one release before flipping the default. At a high level what you say makes sense, but I'm not entirely sure that waiting one release fixes the problem? Meaning, do we codify somewhere that management servers have to prepare for new versions of Envoy within a certain time frame? If we wait a release, wouldn't someone just come back and say we need to wait 2 releases? Etc.? (This might be a fine policy in practice, but I'm saying it mostly for the sake of discussion.) The flip side of this is that once the API is added, a management server could update even pre-release and set the field to false.

So I guess the question is really around what grace period is there? I think that is what needs to be explicitly codified for this case.